src/lib/connectors/shared/remote-debugging-connector.ts

```diff
@@ -186,12 +194,26 @@ export class Connector implements IConnector {
         }
 
         if (params.redirectResponse) {
-            debug(`Redirect found for ${requestUrl}`);
+            debug(`Redirect found for ${cutString(requestUrl)}`);
+
+            if (requestUrl === params.redirectResponse.url) {
+                debug(`Error redirecting: ${requestUrl} is an infinite loop`);
+
+                return;
+            }
+
+            const hops = this._redirects.calculate(requestUrl);
+
+            // We limit the number of redirects
+            if (hops.length >= 10) {
+                return;
```

Comment:
Maybe we should warn to the user that something wrong is happening. The same things with the `debug` above